generators/service/enabler/index.js

```diff
@@ -179,102 +177,94 @@ module.exports = class extends Generator {
 			}
 			serviceCredentials['serviceInfo'] = deployOpts && Object.prototype.hasOwnProperty.call(deployOpts, 'service_bindings') ? deployOpts.service_bindings[this.scaffolderName] : {};
 		}
-		this.logger.debug(`_addMappings - serviceCredentials=${JSON.stringify(serviceCredentials, null, 3)}`);
-		this.logger.debug(`_addMappings - deploy_options=${JSON.stringify(this.context.deploy_options, null, 3)}`);
-		let scaffolderKeys = this._setCredentialMapping({}, serviceCredentials, this.serviceKey);
-		scaffolderKeys = Object.keys(scaffolderKeys).map(key => {
-			let scaffolderKey = key.split(`${this.serviceKey.replace(/-/g, '_')}_`);
-			if (Array.isArray(scaffolderKey) && scaffolderKey.length > 1) {
-				return scaffolderKey[1];
-			}
-		});
-
-		let version = 1;
-		let credentialKeys = scaffolderKeys.filter(key => { return key !== 'serviceInfo' });
-		let credKeysToScaffolderKeysMap = {};
-
-		scaffolderKeys.sort();
-		credentialKeys.sort();
-
-		credKeysToScaffolderKeysMap = this._mapCredentialKeysToScaffolderKeys(credentialKeys, scaffolderKeys);
-
 
-		let mapping = fs.readFileSync(path.join(__dirname, '../templates', `mappings.v${version}.json.template`), 'utf-8');
+		const mappings = this._generateMappingsJson(this.context, this.scaffolderName);
 
-		Handlebars.registerHelper('access', (map, key, nestedJSON) => {
-			return nestedJSON ? map[key].replace('_', '.') : map[key];
-
-		});
+		this.context.addMappings(mappings);
+	}
 
-		let template = Handlebars.compile(mapping);
-		let localServiceKey = this.serviceKey;
-		let serviceKeySeparator = '_'
-		let localCredentialKeys = [];
-		let springMapping = null
-		if (this.context.application.language === "SPRING") {
```

Comment:
Have you tested this against knative eventing? I'm not sure what framework specific formatting is going on here but I could see it breaking that skit.